recipes/remix/app/routes/_index.tsx

```diff
@@ -1,41 +1,50 @@
-import type { MetaFunction } from "@remix-run/node";
+import { Render } from "@measured/puck";
+import styles from "@measured/puck/dist/index.css";
+import type { LoaderFunctionArgs, MetaFunction } from "@remix-run/node";
+import { json, LinksFunction } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
 
-export const meta: MetaFunction = () => {
-  return [
-    { title: "New Remix App" },
-    { name: "description", content: "Welcome to Remix!" },
-  ];
+import puckConfig from "../../puck.config";
+import { getPage } from "~/models/page.server";
+
+/**
+ * Disable client-side JS - Optional
+ * If you know that your Puck content doesn't need react.
+ * Then you can disable JS for this route.
+ * @see https://remix.run/docs/en/main/guides/disabling-javascript
+ */
+
+// export const handle = { hydrate: false };
+
+export const loader = async ({ params }: LoaderFunctionArgs) => {
+  // Get path, and default to slash for root path.
+  const puckPath = params.puckPath || "/";
+  // Get puckData for this path, this could be a database call.
+  const puckData = getPage(puckPath);
+  if (!puckData) {
+    throw new Response(null, {
+      status: 404,
+      statusText: "Not Found",
+    });
+  }
+  // Return the data.
+  return json({ puckData });
+};
+
+export const meta: MetaFunction<typeof loader> = ({ data }) => {
+  const title = data?.puckData?.root?.title || "Untitled page";
```

Comment:
```suggestion
  const title = data?.puckData?.root?.title || "Page";
```

How about this? I think "Untitled" looks worse if users forget to set it.